lib/src/metta/runner/builtin_mods/json.rs

```diff
@@ -0,0 +1,223 @@
+use hyperon_atom::{Atom, ExecError, gnd::GroundedFunctionAtom, ExpressionAtom};
+use crate::metta::{ARROW_SYMBOL, ATOM_TYPE_ATOM, ATOM_TYPE_SPACE};
+use crate::space::grounding::GroundingSpace;
+use crate::metta::text::SExprParser;
+use crate::metta::runner::{Metta, ModuleLoader, RunContext};
+use crate::metta::runner::modules::MettaMod;
+use crate::metta::runner::str::{atom_to_string, ATOM_TYPE_STRING};
+use crate::metta::runner::str::Str;
+use serde_json::Value;
+use crate::metta::runner::DynSpace;
+use crate::metta::runner::bool::*;
+use crate::metta::runner::number::Number;
+use crate::metta::types::{get_atom_types, get_meta_type};
+
+pub static JSON_METTA: &'static str = include_str!("json.metta");
+
+#[derive(Debug)]
+pub(crate) struct JsonModLoader;
+
+impl ModuleLoader for JsonModLoader {
+    fn load(&self, context: &mut RunContext) -> Result<(), String> {
+        // Initialize module's space
+        let space = GroundingSpace::new();
+        context.init_self_module(space.into(), None);
+
+        // Load module's tokens
+        let _ = self.load_tokens(context.module(), context.metta.clone())?;
+
+        // Parse MeTTa code of the module
+        let parser = SExprParser::new(JSON_METTA);
+        context.push_parser(Box::new(parser));
+
+        Ok(())
+    }
+
+    fn load_tokens(&self, target: &MettaMod, _metta: Metta) -> Result<(), String> {
+        let mut tref = target.tokenizer().borrow_mut();
+
+        tref.register_function(GroundedFunctionAtom::new(
+            r"json-encode".into(),
+            Atom::expr([ARROW_SYMBOL, ATOM_TYPE_ATOM, ATOM_TYPE_STRING]),
+            json_encode));
+
+        tref.register_function(GroundedFunctionAtom::new(
+            r"json-decode".into(),
+            Atom::expr([ARROW_SYMBOL, ATOM_TYPE_STRING, ATOM_TYPE_SPACE]),
+            json_decode));
+
+        Ok(())
+    }
+}
+
+fn encode_list(input: Vec<Atom>) -> Result<String, ExecError> {
+    let mut res_str = "".to_string();
+    for atom in input {
+        match encode_atom(&atom) {
+            Ok(encoded) => {res_str = res_str + ", " + &encoded;},
+            Err(err) => return Err(err),
+        }
+    }
+    res_str.remove(0);
+    res_str.remove(0);
+    Ok(format!("[{}]", res_str))
+}
+
+fn encode_dictspace(input: &Atom) -> Result<String, ExecError> {
+    let space = Atom::as_gnd::<DynSpace>(input).unwrap();
+    let result = space.borrow().subst(&Atom::expr([Atom::var("k"), Atom::var("v")]),
+                                      &Atom::expr([Atom::var("k"), Atom::var("v")]));
```

Comment:
I would use `Space::visit` instead, because it should be more effective at the moment for a big dictionaries.